### PR TITLE
Polish `GradleLifecycle` implementation

### PR DESCRIPTION
**Pull request by bamboo**
Follow-up to #28875

### Reviewing cheatsheet

Before merging the PR, comments starting with 
- ❌ ❓**must** be fixed
- 🤔 💅 **should** be fixed
- 💭 **may** be fixed
- 🎉 celebrate happy things

### CODE REVIEW

**Comment by bamboo** [comment]
@bot-gradle test

**Comment by alllex** [comment]
While at it, could you also fix the javadoc for `IsolatedProjectEvaluationListenerProvider.clear` and add the service scope to that interface? I believe it should be `@ServiceScope(Scope.Build.class)`.

Also, related to this service I noticed this: https://github.com/gradle/gradle/pull/28581/files#r1578518821

**Review by alllex — APPROVED** [review]

**Comment by bamboo** [comment]
@bot-gradle merge

**Comment by bamboo** [comment]
Thanks, @alllex!

File: platforms/core-configuration/configuration-cache/src/main/kotlin/org/gradle/configurationcache/extensions/ServiceRegistrationExtensions.kt [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+import org.gradle.internal.service.ServiceRegistration
+
+
+/**
+ * @param ServiceType The service to make visible.
+ * @param ImplementationType The implementation type of the service.
+ * @see [ServiceRegistration.add]
+ */
+internal
+inline fun <reified ServiceType, reified ImplementationType : ServiceType> ServiceRegistration.add() {
```
Comment (alllex) [inline platforms/core-configuration/configuration-cache/src/main/kotlin/org/gradle/configurationcache/extensions/ServiceRegistrationExtensions.kt:28]: 👍

**Pull request merged**